### DISCUSSION TITLE
Error codes in V2 v. V1?

### DISCUSSION THREAD

**Discussion by GanizaniSitara** [tweepy/tweepy#1858] [Questions] [answered]
In V1 I could do 

`print (e.response)`
`print (e.api_errors)`
`print (e.api_codes)`
`print (e.api_messages)`

In V2 the error object doesn't seem to have any properties at all? What am I missing. Looking

**Comment by Harmon758** [comment]
What error object are you referring to?
The same exceptions are used across the library.

**Comment by GanizaniSitara** [comment]
Thanks for the response. This is what I'm seeing:

```
try:
    status = api.update_status(status="test", in_reply_to_status_id = 1514213177210425346 , auto_populate_reply_metadata=True)
except tweepy.errors.Forbidden as e:
    print(e.response)
    print(e.api_errors)
    print(e.api_codes)
    print(e.api_messages)
# <Response [403]>
# [{'code': 433, 'message': 'The original Tweet author restricted who can reply to this Tweet.'}]
# [433]
# ['The original Tweet author restricted who can reply to this Tweet.']
# Deleted or not visible.

try:
    response = client.create_tweet(text="test", in_reply_to_tweet_id=1514213177210425346)
except tweepy.errors.Forbidden as e:
    print(e.response)
    print(e.api_errors)
    print(e.api_codes)
    print(e.api_messages)
# <Response [403]>
# []
# []
# []
```

**Comment by Harmon758** [comment, marked as answer]
Unfortunately, Twitter's API doesn't actually return an error code or message in the response to `Client.create_tweet` with a restricted `in_reply_to_tweet_id`.

Despite what https://developer.twitter.com/en/support/twitter-api/error-troubleshooting says, the response doesn't include an `errors` field:
`{"detail":"You are not allowed to reply to this Tweet as the Tweet author has restricted who can reply to this Tweet.","type":"about:blank","title":"Forbidden","status":403}`

This exemplifies [the issue with inconsistency in error response structures with Twitter API v2 that I encountered when I reworked exceptions in the library while adding support for Twitter API v2](https://twittercommunity.com/t/questions-about-error-response-structure/151746).

Since there seems to be a relevant message in the `detail` field of the response in this case, I've added that as a fallback message for `HTTPException` with https://github.com/tweepy/tweepy/commit/b6b8241d8df408a427a38d3b9a44837f07cfab32.

**Reply by GanizaniSitara** [reply]
That's great, thank you. Tinkering, I came to the same conclusion. Appreciate the response.
